### ISSUE TITLE
TODOs

### ISSUE THREAD

**Issue by omoindrot** [cs230-stanford/cs230-code-examples#1]
## TODO

- [x] change name `eval_metrics` to `metrics`
  - introduces confusion: only have one set of metrics that are said to be an average on the dataset
  - what we log with tqdm is not metrics but "training monitoring"
- [x] enable hyperparameter search
  - `hyperparam_search.py` calling `train.py` over multiple `params.json` in `experiments`
  - `run("train.py --model_dir experiments/exp024")`
- [x] check that there is no problem with the virtual env and that we don't need to add a preamble
   
- [x] saving
  - if we train again the model, training should start where it stopped
  - add an optional argument for this?
 
- [x] split the graph into train and eval
  - clean-up
  - reuse the weights with 2 graphs and 2 inputs
  - (or use a placeholder??)

- [x] split `hyperparams_search.py` into two files?
  - one for hyperparam search
  - one for "syntethizing results"

- [x] for synthesize results, get all possible subdirectories in it
 
- [ ] explore file `explore.py` ?
  - would run the model on some examples (dataset)
  - give access to some errors
  - interaction? (ipython like test?)

- [x] explicitely have a train / dev / test split
  - for the SIGNS dataset, there is only train / test --> do the split in train.py
  - some images are duplicated??? Ignore or clean up the dataset?

- [x] add tf.summary.image for training images?
 
- [x] make sure there are only images in SIGNS (no .DS_Store)
 
- [x] add script `download_data.sh`

- [x] add descriptions to `parser.add_argument(help="...")`

- [x] perform synthesize metrics after hyperparam?

- [x] Use mode (`mode={'train', 'eval', 'predict'}`) instead of `is_training`
  - allow use of predict mode for functions
  - pros: allows prediction (without labels)
  - cons: adds more code

- [x] use restore_from (file or dir)
  - when restoring weights, keep track of the number of epochs already run?

- [x] when restoring, the numbering is becoming incorrect as it starts from 0 again

- [ ] make first argument from `input_fn` be `train` or `eval` instead of `True`/`False`

- [ ] check the date for each blog post (make everything dated as 02/01?)
